doc/user_guide/user_guide.md

```diff
@@ -0,0 +1,291 @@
+python -m exasol_advanced_analytics_framework.deploy language-container \
+    --dsn <DB_HOST:DB_PORT> \
+    --db-user <DB_USER> \
+    --db-pass <DB_PASSWORD> \
+    --bucketfs-name <BUCKETFS_NAME> \
+    --bucketfs-host <BUCKETFS_HOST> \
+    --bucketfs-port <BUCKETFS_PORT> \
+    --bucketfs-user <BUCKETFS_USER> \
+    --bucketfs-password <BUCKETFS_PASSWORD> \
+    --bucket <BUCKETFS_NAME> \
+    --path-in-bucket <PATH_IN_BUCKET> \
+    --language-alias "$LANGUAGE_ALIAS" \
+    --container-file "$SLC_FILE"
+```
+
+### Additional Scripts
+
+Besides the BucketFS connection, the SLC, and the Python package AAF also requires some additional Lua scripts to be created in the Exasol database.
+
+The following command deploys the additional scripts to the specified `DB_SCHEMA` using the `LANGUAGE_ALIAS` of the SLC:
+
+```shell
+python -m exasol_advanced_analytics_framework.deploy scripts \
+    --dsn <DB_HOST:DB_PORT> \
+    --db-user <DB_USER> \
+    --db-pass <DB_PASSWORD> \
+    --schema <DB_SCHEMA> \
+    --language-alias "$LANGUAGE_ALIAS"
+```
+
+## Usage
+
+The entry point of this framework is `AAF_RUN_QUERY_HANDLER` script. This script is simply a query loop which is responsible for executing the implemented algorithm.
+
+This script takes the necessary parameters to execute the desired algorithm in string json format. The json input includes two main part:
+
+* `query_handler` : Details of the algorithm implemented by user.
+* `temporary_output`:  Information about BucketFS where the temporary outputs of the query handler is kept.
+
+The following SQL statement shows how to call an AAF query handler:
+
+```sql
+EXECUTE SCRIPT AAF_RUN_QUERY_HANDLER('{
+    "query_handler": {
+        "factory_class": {
+            "module": "<CLASS_MODULE>",
+            "name": "<CLASS_NAME>"
+        },
+        "parameters": "<CLASS_PARAMETERS>",
+        "udf": {
+            "schema": "<UDF_DB_SCHEMA>",
+            "name": "<UDF_NAME>"
+        }
+    },
+    "temporary_output": {
+        "bucketfs_location": {
+            "connection_name": "<BUCKETFS_CONNECTION_NAME>",
+            "directory": "<BUCKETFS_DIRECTORY>"
+        },
+        "schema_name": "<TEMP_DB_SCHEMA>"
+    }
+}');
+```
+
+See [Implementation of Custom Algorithms](#implementation-of-custom-algorithms) for a complete example.
+
+### Parameters
+
+| Parameter                    | Required? | Description                                                                                                    |
+|------------------------------|-----------|----------------------------------------------------------------------------------------------------------------|
+| `<CLASS_NAME>`               | yes       | Name of the query handler class                                                                                |
+| `<CLASS_MODULE>`             | yes       | Module name of the query handler class <span style="color: red">(should we mention `builtins` here?)</span>    |
```

Comment:
the builtins only comes into play when we implement the queryhandler in UDF. We should mention that in the example.

Reply:
See next push.